src/Rel8/Statement/Where.hs

```diff
@@ -0,0 +1,37 @@
+{-# language MonoLocalBinds #-}
+{-# language NamedFieldPuns #-}
+{-# language StrictData #-}
+
+module Rel8.Statement.Where
+  ( Where, ppWhere
+  )
+where
+
+-- base
+import Data.Functor ( (<&>) )
+import Prelude
+
+-- pretty
+import Text.PrettyPrint ( Doc, ($$), parens, text )
+
+-- rel8
+import Rel8.Expr.Bool ( true )
+import Rel8.Query ( Query )
+import Rel8.Schema.Name ( Selects )
+import Rel8.Schema.Table ( TableSchema(..) )
+import Rel8.Statement.Select ( ppSelect )
+import Rel8.Table.Opaleye ( view )
+
+
+-- | The @WHERE@ condition in a @DELETE@ or @UPDATE@ statement. This can be
+-- an arbitrary 'Query': if the query returns zero rows, then the condition
+-- is false, if the query returns one or more rows, then the condition is
+-- true. Most of the time you'll want to use 'Rel8.where_' to supply an
+-- explicit boolean value.
+type Where expr = expr -> Query ()
+
+
+ppWhere :: Selects names exprs => TableSchema names -> Where exprs -> Maybe Doc
+ppWhere TableSchema {columns} where_ =
+  ppSelect (true <$ where_ (view columns)) <&> \query ->
+    text "WHERE EXISTS" $$ parens query
```

Comment:
Unfortunately I'm not sure this is going to be performant. Let's take the very simple case of `WHERE x.id = 123`. We could have:

```sql
delete from part where part.id = 123
```

or we could have

```sql
delete from part p where exists (select 0 where p.id = 123);
```

I _think_ this is what your PR here would produce, with `deleteWhere = where_ . (lit 123 ==.) . partId`. 

The former has a good plan:

```
Delete on part  (cost=0.08..4.09 rows=1 width=6)
   ->  Index Scan using part_pkey on part  (cost=0.08..4.09 rows=1 width=6)
         Index Cond: ((id)::bigint = 123)
```

But the latter `WHERE EXISTS` approach produces:

```
 Delete on part p  (cost=0.00..186750.21 rows=294554 width=6)
   ->  Seq Scan on part p  (cost=0.00..186750.21 rows=294554 width=6)
         Filter: (alternatives: SubPlan 1 or hashed SubPlan 2)
         SubPlan 1
           ->  Result  (cost=0.00..0.00 rows=1 width=0)
                 One-Time Filter: ((p.id)::bigint = 123)
         SubPlan 2
           ->  Result  (cost=0.00..0.00 rows=1 width=4)
```

I haven't actually tested the PR though, so there's a chance that it doesn't actually produce this SQL. Might be worth digging into this some more.